backend/src/apiserver/server/task_server.go

```diff
@@ -0,0 +1,82 @@
+package server
+
+import (
+	"context"
+	api "github.com/kubeflow/pipelines/backend/api/go_client"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/model"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/resource"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+)
+
+type TaskServer struct {
+	resourceManager *resource.ResourceManager
+}
+
+func (s *TaskServer) CreateTask(ctx context.Context, request *api.CreateTaskRequest) (*api.Task, error) {
+	err := s.validateCreateTaskRequest(request)
+	if err != nil {
+		return nil, util.Wrap(err, "Validate create task request failed.")
+	}
+
+	task, err := s.resourceManager.CreateTask(ctx, request.Task)
+	if err != nil {
+		return nil, util.Wrap(err, "Failed to create a new task.")
+	}
+
+	return ToApiTask(task), nil
+}
+
+func (s *TaskServer) validateCreateTaskRequest(request *api.CreateTaskRequest) error {
+	task := *request.Task
+
+	empty := func(s string) bool { return len(s) == 0 }
```

Comment:
nit: I don't think it's necessary creating an empty function.

`if task.GetId() == ""` is idiomatic Go code and also readable. `empty` will require double checking what the implementation is.